contracts/src/v0.8/dev/keeper2_0/KeeperRegistry2_0.sol

```diff
@@ -191,192 +222,160 @@ contract KeeperRegistry2_0 is
-    _fallback();
-  }
+  ////////
+  // SETTERS
+  ////////
 
   /**
-   * @notice proposes the safe transfer of a keeper's payee to another address
-   * @param keeper address of the keeper to transfer payee role
-   * @param proposed address to nominate for next payeeship
+   * @inheritdoc OCR2Abstract
    */
-  function transferPayeeship(address keeper, address proposed) external {
-    // Executed through logic contract
-    _fallback();
-  }
+  function setConfig(
+    address[] memory signers,
+    address[] memory transmitters,
+    uint8 f,
+    bytes memory onchainConfig,
+    uint64 offchainConfigVersion,
+    bytes memory offchainConfig
+  ) external override onlyOwner {
+    if (signers.length > maxNumOracles) revert TooManyOracles();
+    if (f == 0) revert IncorrectNumberOfFaultyOracles();
+    if (signers.length != transmitters.length || signers.length <= 3 * f) revert IncorrectNumberOfSigners();
+    if (onchainConfig.length != 0) revert OnchainConfigNonEmpty();
 
-  /**
-   * @notice accepts the safe transfer of payee role for a keeper
-   * @param keeper address to accept the payee role for
-   */
-  function acceptPayeeship(address keeper) external {
-    // Executed through logic contract
-    _fallback();
-  }
+    // remove any old signer/transmitter addresses
+    uint256 oldLength = s_signersList.length;
+    address signer;
+    address transmitter;
+    for (uint256 i = 0; i < oldLength; i++) {
+      signer = s_signersList[i];
+      transmitter = s_transmittersList[i];
+      delete s_signers[signer];
+      // Do not delete the whole transmitter struct as it has balance information stored
+      s_transmitters[transmitter].active = false;
+    }
+    delete s_signersList;
+    delete s_transmittersList;
 
-  /**
-   * @notice proposes the safe transfer of an upkeep's admin role to another address
-   * @param id the upkeep id to transfer admin
-   * @param proposed address to nominate for the new upkeep admin
-   */
-  function transferUpkeepAdmin(uint256 id, address proposed) external override {
-    // Executed through logic contract
-    _fallback();
-  }
+    // add new signer/transmitter addresses
+    for (uint256 i = 0; i < signers.length; i++) {
+      if (s_signers[signers[i]].active) revert RepeatedSigner();
+      s_signers[signers[i]] = Signer({active: true, index: uint8(i)});
 
-  /**
-   * @notice accepts the safe transfer of admin role for an upkeep
-   * @param id the upkeep id
-   */
-  function acceptUpkeepAdmin(uint256 id) external override {
-    // Executed through logic contract
-    _fallback();
-  }
+      if (s_transmitters[transmitters[i]].active) revert RepeatedTransmitter();
+      s_transmitters[transmitters[i]].active = true;
+      s_transmitters[transmitters[i]].index = uint8(i);
+    }
+    s_signersList = signers;
+    s_transmittersList = transmitters;
+    s_hotVars.f = f;
+    s_offchainConfigVersion = offchainConfigVersion;
+    s_offchainConfig = offchainConfig;
```

Comment:
as discussed on our call, this is very pricey

Reply:
removed